test/functional/plugin/lsp_spec.lua

```diff
@@ -813,6 +813,74 @@ describe('LSP', function()
     end)
   end)
 
+  describe('apply_text_document_edit', function()
+    before_each(function()
+      insert(dedent([[
+        First line of text
+        Second line of text]]))
+    end)
+    it('correctly goes ahead with the edit when all is normal', function()
+      local text_document_edit = {
+        edits = {
+          range = {
+            start = { line = 1, character = 1 };
+            ["end"] = { line = 1, character = 2 };
+          },
+          newText = "bar"
+        },
+        textDocument = {
+          uri = "/fake/uri";
+          version = 5
+        }
+      }
+      exec_lua('vim.lsp.util.apply_text_document_edit(...)', text_document_edit, 1)
```

Comment:
So I'm stuck here on this will fail with the following:

```
util.lua:163: attempt to compare number with nil
```
Since I'm pretty sure that this "buffer" isn't versioned in the test. I don't really get how to version the buffer for each test. Any pointers?

Reply:
have you tried my recommandation from https://github.com/neovim/neovim/pull/12185#issuecomment-619617312 ?

Reply:
Yes, when adding in this before the line above, I get the same message.

```
exec_lua('luado vim.lsp.util.buf_versions={4,5}')
```
But to be honest, I'm not even sure if I'm using this correctly.

Reply:
the problem comes from the URI `uri = "/fake/uri";` it will be converted to a bufnr  via local `bufnr = vim.uri_to_bufnr(text_document.uri)`. This  call fails in some way, thus `M.buf_versions[bufnr]` fails.
`exec_lua('vim.lsp.util.buf_versions={4,5}')` should be enough.

Reply:
I don't understand where that command is coming from. When I do `:h lsp.util.buf_versions` I get nothing back. Also I understand that the `uri` may be the problem, but in order to create a valid `textDocument` interface I need one. If not, then `uri.lua` blows up since it's expecting a string and gets `nil`.

Reply:
`/fake/uri` is not valid uri.
Try to use `file://fake/uri`.

Reply:
Ah, of course, that makes sense for the uri. I changed that, and now I'm finally back to the initial position with the:

```
test/functional/helpers.lua:106: Error executing lua: ...uments/lua-workspace/neovim/runtime/lua/vim/lsp/util.lua:163: attempt to compare number with nil
```

As I said before it's because the buffer number isn't set. However, I don't really understand the 
```
exec_lua('vim.lsp.util.buf_versions={4,5}')
```

Reply:
Try printing the value returned by `bufnr = vim.uri_to_bufnr(text_document.uri)`. 
`vim.lsp.util.buf_versions={4,5}` fills the buf_versions with a list such that `buf_versions[1]= 4 buf_versions[2]=5`. If the printed value is not in that range 1-2 than the call would fail.